### PR TITLE
idl: record full positions for constants

### PR DESCRIPTION
**Pull request by jparise**
We now have enough information to record full positions (line+column)
for constant values.

### CODE REVIEW

File: idl/info.go [hunk 1]
```diff
@@ -35,6 +35,5 @@ func (i *Info) Pos(n ast.Node) ast.Position {
 	if pos, ok := ast.Pos(n); ok {
 		return pos
 	}
-	pos := i.nodePositions[n]
-	return ast.Position{Line: pos.Line}
+	return i.nodePositions[n]
```
Comment (jparise) [inline idl/info.go:38]: This change should have been part of d7c4d71abe562dc7852dac633a84e1c8130f0985 (#505).

**Comment by jparise** [comment]
The coverage drop doesn't seem accurate. It appears to be reacting to the fact that I replaced two existing lines with one, which is skewing the math.

**Review by abhinav — APPROVED** [review]
Thanks!

When we add position information to the other ast.Nodes, we might be able to drop the separate `lineno` component completely. (Or have it use pos to get the line number.)

**Comment by jparise** [comment]
> When we add position information to the other ast.Nodes, we might be able to drop the separate `lineno` component completely. (Or have it use pos to get the line number.)

Right. I've started to make progress in that direction locally. Just looking for a little direction here: https://github.com/thriftrw/thriftrw-go/issues/349#issuecomment-871003807

**Pull request merged**